eng/resolveContract.targets

```diff
@@ -17,7 +20,13 @@
   <!-- intentionally empty since we no longer need a target -->
   <Target Name="ResolveMatchingContract" />
 
-  <Target Name="VerifyMatchingContract" AfterTargets="ResolveReferences">
-    <Error Condition="'$(HasMatchingContract)' == 'true' and !Exists('%(ResolvedMatchingContract.Identity)')" Text="ResolveMatchingContract could not find a matching contract '%(ResolvedMatchingContract.Identity)' not found." />
+  <!-- Disable matching ref API compat for aot runtime refencing libs, these have a lot of public API
+         only by implementation and we don't intend to start new API development on AOT -->
+  <Target Name="DisableRunMatchingRefApiCompat" 
```

Comment:
Just wondering, why does this have to happen via a Target and not defined statically like for non-aot libraries.

Reply:
I wanted to match the condition that was previously used where this is disabled when `@(ReferenceFromRuntime)` is used.  Since ReferenceFromRuntime is an item and we need to set a property based on its presence it must be done in a Target.  I could have conditioned on "IsPartialFacade" instead but that would disable APICompat in more cases than are necessary.  I could have just ignored this and instead added baselines as we did for non-aot.  I decided not to do so because right now I'm trying to make AOT less of a tax on this repo until we decide how were going to unify it with the 3.0 codebase.